### PR TITLE
Profile-page and preference changes

### PR DESCRIPTION
**Pull request by js1998**

### CODE REVIEW

File: imports/redux/reducers/index.jsx [hunk 1]
```diff
@@ -142,11 +150,30 @@ function userReducer(state = initialUserState, action) {
...
       return { ...state, favourites: info.profile.preferences.favourites };
     case REMOVE_FAVOURITES:
-      //TODO: change this to use MongoToDelete
-      let updatedFavourites = Array.filter((value, index, array) => {
-        return action.favouriteToRemove !== value;
-      })
-      return { ...state, favourites: updatedFavourites };
+      matchedUsers = Meteor.users.update({_id: state.userId}, {$pull:{"profile.preferences.favourites": action.favouriteToRemove}})
+      if (matchedUser === 0){
+        console.log("Error Removing From Favourites")
```
Comment (hailin-zhang) [inline imports/redux/reducers/index.jsx]: remove console logs pls

File: imports/redux/reducers/index.jsx [hunk 2]
```diff
@@ -142,11 +150,30 @@ function userReducer(state = initialUserState, action) {
...
+      }
+      updatedInfo = Meteor.users.find({_id: state.userId}).fetch();
+      info = updatedInfo[0];
+      return { ...state, favourites: info.profile.preferences.favourites };
+    case SAVE_PREVIOUS_TRAVEL:
+      //may not be able to save to meteor a javascript object -- TEST THIS
+      matchedUsers = Meteor.users.update({_id: state.userId}, {$push:{"profile.previousTravels": action.prevTravel}})
+      if (matchedUsers == 0){
+        console.log("Error Saving Travel")
```
Comment (hailin-zhang) [inline imports/redux/reducers/index.jsx]: should you use `console.error` instead?

File: imports/redux/reducers/index.jsx [hunk 3]
```diff
@@ -142,11 +150,30 @@ function userReducer(state = initialUserState, action) {
...
+      }
+      updatedInfo = Meteor.users.find({ userId: state.userId}).fetch();
+      info = updatedInfo[0];
+      return { ...state, previousTravels: info.profile.previousTravels };
+    case DELETE_PREVIOUS_TRAVEL:
+      matchedUsers = Meteor.users.update({_id: state.userId}, {$pull:{"profile.previousTravels": action.toDeleteTravel}})
+      if (matchedUsers == 0){
+        console.log("Error Deleting Travel")
```
Comment (hailin-zhang) [inline imports/redux/reducers/index.jsx]: console.error?

File: imports/ui/pages/profile/components/previous-travels-container.jsx [hunk 4]
```diff
@@ -0,0 +1,33 @@
...
+class PreviousTravelContainer extends React.Component {
+
+    getBlacklist = () => {
+        return Array.from(this.props.previousTravels).map((value, index) => {
+            return (
+              <PreferenceButton key={index} name={value} type={DELETE_PREVIOUS_TRAVEL}/>
+            );
+        }
+    )
+}
```
Comment (hailin-zhang) [inline imports/ui/pages/profile/components/previous-travels-container.jsx]: your spacing is very weird, can you try to follow the styling in other pages?

Comment (hailin-zhang) [inline imports/ui/pages/profile/components/previous-travels-container.jsx]: make sure this bracket lines up with the one on line 9

File: imports/ui/pages/profile/components/previous-travels-container.jsx [hunk 5]
```diff
@@ -0,0 +1,33 @@
...
+
```
Comment (hailin-zhang) [inline imports/ui/pages/profile/components/previous-travels-container.jsx]: 1 line whitespace between methods

File: imports/ui/pages/profile/components/previous-travels-container.jsx [hunk 6]
```diff
@@ -0,0 +1,33 @@
...
+    render() {
+        return (
+          <div className="previous-travels-container">
+            <br/>
+          </div>
+        );
+    }
+}
```
Comment (hailin-zhang) [inline imports/ui/pages/profile/components/previous-travels-container.jsx]: make sure this lines up with the squiggly bracket on line 18

File: imports/ui/pages/profile/profile.page.css [hunk 7]
```diff
@@ -1,83 +1,206 @@
...
-  width: 850px;
-  height: 550px;
-  left: 90px;
-  top: 200px;
-  background: #EBEBEB;
-  border-radius: 10px;
-}
+  .profile {
+    position: relative;
+    width: 850px;
```
Comment (hailin-zhang) [inline imports/ui/pages/profile/profile.page.css]: as a general rule of thumb: for responsive design, you never want to hard-code by pixels!

Comment (hailin-zhang) [reply imports/ui/pages/profile/profile.page.css]: See my review comments at the bottom for how to fix this

File: imports/ui/shared_components/preferences/preference-button.jsx [hunk 8]
```diff
@@ -1,15 +1,44 @@
...
+import { connect } from 'react-redux';
 import './preference-button.css';
 
 class PreferenceButton extends React.Component {
-    render() {
+
+      handleDelete = (type, name) => {
+        switch (type) {
+          case REMOVE_BLACKLIST:
+            console.log("BlackList Case");
```
Comment (hailin-zhang) [inline imports/ui/shared_components/preferences/preference-button.jsx]: remove console logs

File: imports/redux/reducers/index.jsx (hunk 1 above)
Comment (js1998) [reply imports/redux/reducers/index.jsx]: Its only a placeholder until we have something that can handle the error better. I can make a comment and change it to a console.error instead

Comment (hailin-zhang) [reply imports/redux/reducers/index.jsx]: thats fine then! can you add a TODO for this 👍

File: imports/ui/pages/profile/profile.page.css [hunk 9]
```diff
@@ -1,83 +1,241 @@
...
-  font-size: 15px;
+  .profile-blacklist-container {
+    position: relative;
+    width: 96%;
+    height: 25%;
+    margin-top: 1%;
+    margin-left: 2%;
+    float: left;
+    background: #C4C4C4;
+    border-radius: 3px;
```
Comment (hailin-zhang) [inline imports/ui/pages/profile/profile.page.css]: You can change this to `(3 / 1440) * 100vw`

**Comment by pwesferguson** [comment]
> Can you see if you can download the eslint extension and try running it? Otherwise I think we're going to get into a linting back-and-forth that will muddle up PRs. Just the auto-fixable stuff should be fine.

Actually, you can just merge and I'll make a separate linting PR since I didn't do it for all files yet anyway

File: imports/ui/shared_components/navbar/navbar.jsx [hunk 10]
```diff
@@ -34,6 +34,14 @@ class Navbar extends React.Component {
         }, 1400);
       }
 
+    goToProfilePage = async () => {
+      // this.loadingSpinner.current.style.display = 'block';
```
[the remark this answers is not in the document]
Comment (js1998) [reply imports/ui/shared_components/navbar/navbar.jsx]: Donedies

**Pull request merged**